src/relay/pass/enable_tensorrt.cc

```diff
@@ -0,0 +1,585 @@
+}
+
+std::vector<int> GetShape(const Type& type) {
+  const auto* ttype = type.as<TensorTypeNode>();
+  CHECK(ttype);
+  std::vector<int> _shape;
+  for (size_t i = 0; i < ttype->shape.size(); ++i) {
+    auto* val = ttype->shape[i].as<IntImm>();
+    CHECK(val);
+    _shape.push_back(val->value);
+  }
+  return _shape;
+}
+
+class TrtChecker : public ExprVisitor {
+ public:
+  explicit TrtChecker(const std::tuple<int, int, int>& trt_version)
+      : trt_version_(trt_version) {
+    // Create set of compatible ops for this version.
+    trt_compatible_ops.insert(trt_base_compatible_ops.begin(),
+                              trt_base_compatible_ops.end());
+    if (TrtVersionGe(trt_version_, 5, 1, 5)) {
+      // Add TRT 5.1.5 ops to whitelist.
+      trt_compatible_ops.insert(trt_5_1_5_compatible_ops.begin(),
+                                trt_5_1_5_compatible_ops.end());
+    }
+    if (TrtVersionGe(trt_version_, 6, 0, 1)) {
+      // Add TRT 6.0.1 ops to whitelist.
+      trt_compatible_ops.insert(trt_6_0_1_compatible_ops.begin(),
+                                trt_6_0_1_compatible_ops.end());
+    }
+  }
+
+  void VisitExpr_(const VarNode* op) {
+    const auto* ttype = op->checked_type().as<TensorTypeNode>();
+    CHECK(ttype);
+    if (!ttype->dtype.is_float()) {
+      compatible_ = false;
+    }
+  }
+
+  void VisitExpr_(const CallNode* call) final {
+    const std::string op_name = (call->op.as<OpNode>())->name;
+    for (size_t i = 0; i < call->args.size(); ++i) {
+      // Workaround for check for transpose of weight.
+      if ((op_name == "nn.conv2d" || op_name == "nn.dense") && i == 1) {
+        if (auto* transpose = call->args[i].as<CallNode>()) {
+          if (transpose->op.as<OpNode>()->name == "transpose") {
+            if (!transpose->args[0].as<VarNode>()) {
+              compatible_ = false;
+              LOG(INFO) << op_name
+                        << " not supported: most have constant weight.";
+            }
+          } else {
+            compatible_ = false;
+            LOG(INFO) << op_name
+                      << " not supported: most have constant weight.";
+          }
+        } else {
+          VisitExpr(call->args[i]);
+        }
+      } else {
+        VisitExpr(call->args[i]);
+      }
+    }
+    if (trt_compatible_ops.count(op_name) == 0) {
+      LOG(INFO) << op_name << " not supported.";
+      compatible_ = false;
+    }
+    if (op_name == "nn.conv2d") {
+      if (const auto* conv2d_attr = call->attrs.as<Conv2DAttrs>()) {
+        if (conv2d_attr->data_layout != "NCHW" ||
+            conv2d_attr->kernel_layout != "OIHW") {
+          compatible_ = false;
+          LOG(INFO) << op_name << " not supported: must be NCHW.";
+        }
+        if (conv2d_attr->out_layout != "" &&
+            conv2d_attr->out_layout != "NCHW") {
+          compatible_ = false;
+          LOG(INFO) << op_name << " not supported: must be NCHW.";
+        }
+      }
+    }
+    if (op_name == "nn.dense") {
```

Comment:
else if

Reply:
I would suggest we have some helpers to handle each op or each type of ops. It looks to me the readability of this bulk of ifelse stmts is not quite good